add docs for specname

tlambert03 opened the pull request:
assuming you choose to merge #251, this PR adds a section to your docs about hook implementation naming and the `specname` option.

tlambert03 commented on docs/index.rst:
```diff
@@ -268,6 +268,35 @@ then the *hookimpl* should be marked with the ``"optionalhook"`` option:
+same name as the function being decorated by ``@hookimpl`` (e.g.
+``setup_project`` in the example above).
+
+*new in version 0.13.2:*
```
obviously, took a guess at this... feel free to directly edit this (or any other part of this PR).

goodboy commented on docs/index.rst:
```diff
@@ -268,6 +268,35 @@ then the *hookimpl* should be marked with the ``"optionalhook"`` option:
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+During plugin :ref:`registration <registration>`, hook implementations are
+matched to a hook :ref:`specifications <specs>` in the *host* program with the
```
matched to a hook :ref:`specifications <specs>` **if provided** in the *host*..

tlambert03 replied:
to make the sentence read a bit better, I broke into two sentences, restated that there is no string 1-1 requirement, and linked to "enforcing spec validation".

goodboy commented on docs/index.rst:
```diff
@@ -268,6 +268,35 @@ then the *hookimpl* should be marked with the ``"optionalhook"`` option:
+*new in version 0.13.2:*
+
+To override the default behavior, a *hookimpl* may also be matched to a
+*hookspec* in the ``host program`` with a non-matching name by using the
```
non-matching **function** name

goodboy commented on docs/index.rst:
```diff
@@ -268,6 +268,35 @@ then the *hookimpl* should be marked with the ``"optionalhook"`` option:
+To override the default behavior, a *hookimpl* may also be matched to a
+*hookspec* in the ``host program`` with a non-matching name by using the
+``specname`` option.  Continuing the example above, the *hookimpl* function
+does not need to be named ``setup_project``:
```
does not need to be named ``setup_project`` but if `specname="setup_project"` it will be matched and checked against the `setup_project` *hookspec*.

tlambert03 commented:
> I added a few comments. Feel free to push back if I'm being too pedantic 😉

these all work for me!  thanks, added (made a  slight modification to the **if provided** sentence... but I think it's the same sentiment).

you ok with it stating *new in version 0.13.2:*?  is that a safe assumption?

tlambert03 commented:
test failed due to HTTP error.  closing and reopening to trigger new build.
